### PR TITLE
feat(#1337) Concurrency tests for ParseMojo and OptimizeMojo

### LINKED ISSUE #1337: SynchronizedTojo.java:31-34: Replace this custom {@link...
The puzzle `1230-0c34d341` from #1230 has to be resolved: 

https://github.com/objectionary/eo/blob/2fd90f407b6a33a8679a8575314d9659932d968e/eo-maven-plugin/src/main/java/org/eolang/maven/SynchronizedTojo.java#L31-L34

The puzzle was created by rultor on 14-Oct-22. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
I've added several tests in order to fix #1337 issue. The tests help me to detect whether some solution helps or not. I still haven't found the correct way to fix race condition with `Tojo` and `Mono` classes, but it's just the first step. Moreover I had to fix `FakeMaven` in order to make it work with several eo programs.

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@mximp Please, take a look

File: eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java [hunk 1]
```diff
@@ -181,16 +186,41 @@ public Map<String, Path> result() throws IOException {
...
      * Adds eo program to a workspace.
-     * @param path Relative path where to save EO program
      * @param content EO program content.
      * @return The same maven instance.
      * @throws IOException If method can't save eo program to the workspace.
      */
-    private FakeMaven withProgram(final Path path, final String content) throws IOException {
+    private FakeMaven withProgram(final String content) throws IOException {
+        final Path path = Paths.get(
+            String.format("foo/x/main%s.eo", FakeMaven.suffix(this.current.get()))
```
Comment (mximp) [inline eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java:215]: @volodya-lombrozo with concurrent execution we still can get the same number here.
We also don't have any synchronization between `concurrent.get()` and `concurrent.incrementAndGet()` so some other thread might increment the value and foreign catalog might get wrong file written.

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java:215]: @mximp `FakeMaven` isn't thread safe for now and it doesn't have to be. The idea is to create single `FakeMaven` and work with it. I can't imagine the situation where I have to prepare environment in parallel. 
If you insist, I can create a puzzle, but I don't see the reason for it. What do you think?

Comment (mximp) [reply eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java:215]: @volodya-lombrozo Can we mark `FakeMave` as not thread-safe in JavaDoc please

**Review by mximp — COMMENTED** [review]
@volodya-lombrozo please check my comment

File: eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java (hunk 1 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java:217]: @mximp Done

File: eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java [hunk 2]
```diff
@@ -44,6 +45,7 @@
  * @since 0.28.12
  */
 @SuppressWarnings("PMD.TooManyMethods")
+@NotThreadSafe
```
Comment (mximp) [inline eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java:48]: @volodya-lombrozo we normally place comment in JavaDoc `NOT thread-safe`.
If we are to use this new annotation we need to change it in all other places. I don't think it's reasonable in this case.

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java:49]: @mximp `NOT thread-safe` is used only several times :) I've added `NOT thread-safe` to java doc. Please, check.

**Comment by volodya-lombrozo** [comment]
@yegor256 merge please

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**